engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/ComplexType.scala

```diff
@@ -110,12 +108,25 @@ case object ComplexType extends IRPass {
     * @param typ the type definition to desugar
     * @return the top-level definitions corresponding to the desugaring of `typ`
     */
-  def desugarComplexType(
+  private def desugarComplexType(
     typ: IR.Module.Scope.Definition.SugaredType
   ): List[IR.Module.Scope.Definition] = {
-    val annotations = typ.getMetadata(ModuleAnnotations)
+    val annotations     = typ.getMetadata(ModuleAnnotations)
+    var lastAnnotations = Seq.empty[IR.Name.GeneralAnnotation]
+    var seenAnnotations = Set.empty[IR.Name.GeneralAnnotation]
     val atomDefs = typ.body
-      .collect { case d: IR.Module.Scope.Definition.Data => d }
+      .flatMap {
+        case ann: IR.Name.GeneralAnnotation =>
+          lastAnnotations :+= ann
+          None
+        case d: IR.Module.Scope.Definition.Data =>
+          val res = Some(d.copy(annotations = d.annotations ++ lastAnnotations))
+          lastAnnotations.foreach(seenAnnotations += _)
```

Comment:
```suggestion
          seenAnnotations ++= lastAnnotations
```
should work as well?